### PR TITLE
stream: RST no longer acks all data

### PR DESCRIPTION
**Pull request by victorjulien**
Since forever (1578ef1e3e8a24d0cc615430c4e6bec1fefdad28) a valid RST would update the internal `last_ack` representation to include all unack'd data. This was originally done to make sure the unACK'd data was inspected/processed at flow timeout.

It was observed however, that if GAPs existed in this unACK'd data, a GAP could be reported in the stats and a GAP event would be raised. This doesn't make sense, as missing segments in the unACK'd part of the stream are completely normal. Segments simply do not all arrive in order.

It turns out that the original behavior of updating `last_ack` to include all unACK'd data is no longer needed. Both raw stream inspection and app-layer updates will include the unACK'd data on stream timeout.

Since the GAP detection uses `last_ack` to determine GAPs, not moving `last_ack` addresses the GAP false positives.

Ticket: #7422.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/2154

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW1_files_sha256.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPR1_stats_chk | | |
| .flow.end.tcp_liberal | 13588 | 12288 | 90.43% |
| .app_layer.error.http.parser | 700 | 561 | 80.14% |
| .app_layer.error.tls.gap | 1718 | 1649 | 95.98% | 

Pipeline 23616

**Comment by victorjulien** [comment]
@ct0br0 I have the pcaps, no action needed at this point.

**Comment by victorjulien** [comment]
After investigating the missing hashes, my conclusion is that the behavior in this PR is correct. They all seem to have the following pattern:
1. unACK'd data in flight, with some GAPs (some segments missing)
2. RST comes in, `last_ack` doesn't cover all in flight segments

In `master`, the RST will auto-ACK all data, which means we're now sending this incomplete data to the app-layer parser. The parsers get confused and GAPs are counted. The HTTP parser specifically, uses some of the post-GAP data from the in-flight section to create new complete nonsense transaction, with no URI, etc. It's it this nonsense tx that is treated like a file, and for which the hashes are reported missing for this PR.

Since the issues are with TLPW1, I'm crafting some additional SV tests to show the issue.

**Comment by victorjulien** [comment]
https://github.com/OISF/suricata-verify/pull/2154/commits/b7cf9870be47c0b6cd903ce79374e52982227622 has a couple of tests to show the issue with TLPW1.

File: src/stream-tcp.c [hunk 1]
```diff
@@ -3242,10 +3216,9 @@ static int StreamTcpPacketStateEstablished(
             ssn->client.window = window << ssn->client.wscale;
 
             if ((tcph->th_flags & TH_ACK) && StreamTcpValidateAck(ssn, &ssn->server, p) == 0)
-                StreamTcpUpdateLastAck(
-                        ssn, &ssn->server, StreamTcpResetGetMaxAck(&ssn->server, window));
```
Comment (victorjulien) [inline src/stream-tcp.c:3246]: also, notice this fix here ;)

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW1_files_sha256.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPR1_stats_chk | | |
| .flow.end.tcp_liberal | 13588 | 12288 | 90.43% |
| .app_layer.error.http.parser | 700 | 561 | 80.14% |
| .app_layer.error.tls.gap | 1718 | 1649 | 95.98% | 

Pipeline 23719

**Comment by victorjulien** [comment]
Replaced by #12300

**Pull request closed without merging**